prow/scripts/provision-vm-and-start-cluster-users-minikube.sh

```diff
@@ -0,0 +1,120 @@
+#!/bin/bash
+
+# This script is designed to provision a new vm and start cluster-users tests. It takes an optional positional parameter using --image flag
+# Use this flag to specify the custom image for provisining vms. If no flag is provided, the latest custom image is used.
+
+set -o errexit
+
+date
+
+readonly SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+readonly TEST_INFRA_SOURCES_DIR="$(cd "${SCRIPT_DIR}/../../" && pwd)"
+
+export USE_ALPHA=${USE_ALPHA:-false}
+
+# shellcheck source=prow/scripts/lib/gcp.sh
+source "${TEST_INFRA_SOURCES_DIR}/prow/scripts/lib/gcp.sh"
+# shellcheck source=prow/scripts/lib/log.sh
+source "${TEST_INFRA_SOURCES_DIR}/prow/scripts/lib/log.sh"
+# shellcheck source=prow/scripts/lib/utils.sh
+source "${TEST_INFRA_SOURCES_DIR}/prow/scripts/lib/utils.sh"
+
+if [[ "${BUILD_TYPE}" == "pr" ]]; then
+    log::info "Execute Job Guard"
+    "${TEST_INFRA_SOURCES_DIR}/development/jobguard/scripts/run.sh"
+fi
+
+cleanup() {
+    ARG=$?
+    log::info "Removing instance kyma-integration-test-${RANDOM_ID}"
+    gcloud compute instances delete --zone="${ZONE}" "kyma-integration-test-${RANDOM_ID}" || true ### Workaround: not failing the job regardless of the vm deletion result
+    date
+    exit $ARG
+}
+
+function testCustomImage() {
+    CUSTOM_IMAGE="$1"
+    IMAGE_EXISTS=$(gcloud compute images list --filter "name:${CUSTOM_IMAGE}" | tail -n +2 | awk '{print $1}')
+    if [[ -z "$IMAGE_EXISTS" ]]; then
+        log::error "${CUSTOM_IMAGE} is invalid, it is not available in GCP images list, the script will terminate ..." && exit 1
+    fi
+}
+
+gcp::authenticate \
+  -c "${GOOGLE_APPLICATION_CREDENTIALS}"
+
+RANDOM_ID=$(openssl rand -hex 4)
+
+LABELS=""
+if [[ -z "${PULL_NUMBER}" ]]; then
+    LABELS=(--labels "branch=$PULL_BASE_REF,job-name=kyma-integration")
+else
+    LABELS=(--labels "pull-number=$PULL_NUMBER,job-name=kyma-integration")
+fi
+
+POSITIONAL=()
+while [[ $# -gt 0 ]]
+do
+
+    key="$1"
+
+    case ${key} in
+        --image)
+            IMAGE="$2"
+            testCustomImage "${IMAGE}"
+            shift
+            shift
+            ;;
+        --*)
+            echo "Unknown flag ${1}"
+            exit 1
+            ;;
+        *)    # unknown option
+            POSITIONAL+=("$1") # save it in an array for later
+            shift # past argument
+            ;;
+    esac
+done
+set -- "${POSITIONAL[@]}" # restore positional parameters
+
+
+if [[ -z "$IMAGE" ]]; then
+    log::info "Provisioning vm using the latest default custom image ..."
```

Comment:
This is a bit misleading, because it's not provisioning here, but only looks for an image.
I would move this log entry after line 89 and change it to something like that:
log::info "Found a default custom image to provision VM: $IMAGE"